Mathlib/Data/Polynomial/Basic.lean

```diff
@@ -1188,7 +1185,6 @@ instance ring : Ring R[X] :=
       toFinsupp_mul toFinsupp_neg toFinsupp_sub (fun _ _ => toFinsupp_smul _ _)
       (fun _ _ => toFinsupp_smul _ _) toFinsupp_pow (fun _ => rfl) fun _ => rfl with
     toSemiring := Polynomial.semiring,
-    toIntCast := Polynomial.intCast
```

Comment:
Why was this deleted?

Reply:
because I had to rename `Polynomial.intCast` to `Polynomial.instIntCast` and there's no need to specify the instance anyway since `Ring` extends `IntCast`

Reply:
Sure but you get a different path to `IntCast` here. Instead of a direct call you now have to go through all of `Function.Injective.ring` to get to it. The synthesized instance doesn't take priority and it looks like this was rewritten to avoid using anything but data from the `Function.Injective.ring` call

Reply:
I think #11521 will make this moot